cpp/src/barretenberg/stdlib/primitives/group/group.hpp

```diff
@@ -39,8 +42,18 @@ auto group<ComposerContext>::fixed_base_scalar_mul_g1(const field_t<ComposerCont
 
 template <typename ComposerContext>
 template <size_t num_bits>
-auto group<ComposerContext>::fixed_base_scalar_mul(const field_t<ComposerContext>& in, const size_t generator_index)
+point<ComposerContext> group<ComposerContext>::fixed_base_scalar_mul(const field_t<ComposerContext>& in,
+                                                                     const size_t generator_index)
 {
+    if constexpr (IsSimulator<ComposerContext>) {
+        if (in.get_value() == barretenberg::fr(0)) {
+            in.context->failure("input scalar to fixed_base_scalar_mul_internal cannot be 0");
+            return { 0, 0 }; // WORKTODO: std::optional?
+        }
+        auto result = crypto::generators::fixed_base_scalar_mul<num_bits>(in.get_value(), generator_index);
```

Comment:
It _could_ be possible to not just default to the crypto implementation, but instead to execute the circuit implementation without storing intermediate witnesses. That has the benefit of further stress testing the lower-lying stdlib primitives, while this ill be faster and was easier to implement. I do worry about subtle simulator bugs that are hard to debug when beaking the link between the higher-level stuff (hashes, signatures, etc) from the lower level primitices (field elements, uints, etc). Then again. the _same_ code will execute for circuits, so you would pick up such bugs when it matters most.